### DISCUSSION TITLE
Customize DuckDB's user agent string for remote requests

### DISCUSSION THREAD

**Discussion by dbreunig** [duckdb/duckdb#15078] [Ideas]
We find the `ATTACH` statement's ability to use remote databases over HTTP very useful for tutorials and testing. Especially if the remote database is filled with helpful views and other structure, pointing at remote datasets letting us keep our initial database size down.

It would be great if the URL provided could be the root URL, allowing the HTTP server to detect the request as one from DuckDB and respond with the database attachment. [Here's an example of how you'd achieve this today](https://www.dbreunig.com/2024/12/01/turning-your-root-url-into-a-duckdb-remote-database.html), using the fairly unique (if most of your hits are browsers) current user agent string that is the default of cpp-httplib.

Having a user agent that contains 'duckdb' and the current version make this feature much more robust.

**Comment by dbreunig** [comment]
Just discovered the user agent when installing extensions is `duckdb/v1.1.3(osx_arm64) cli 19864453f7`. Seems like an inconsistency that could be fixed.

**Reply by mlafeldt** [reply]
I opened https://github.com/duckdb/duckdb/issues/15095 mere minutes ago. 😄

**Reply by dbreunig** [reply]
Nice, thanks.

**Comment by soerenwolfers** [comment]
Wouldn't it be cleaner if DuckDB registered `application/vnd.duckdb` at IANA and then sent a `Accept: application/vnd.duckdb` header in its initial `ATTACH` `GET` request (maybe only when there is no file extension) and only sent a follow-up `Accept: */*` if it gets a 415 response?

**Reply by soerenwolfers** [reply]
relying on the user agent sounds wrong. After all, you do *not* want to respond the same to an ATTACH request as to an INSTALL

**Reply by dbreunig** [reply]
100%, having `application/vnd.duckdb` would be the ideal. But even then: a specific and consistent user agent would remain helpful. I recommend both paths, fixing the user case first.
